examples/object_detection/ios/ObjectDetector/Managers/ObjectDetectorService.swift

```diff
@@ -0,0 +1,253 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import MediaPipeTasksVision
+import AVFoundation
+
+/**
+ This protocol must be adopted by any class that wants to get the detection results of the object detector in live stream mode.
+ */
+protocol ObjectDetectorServiceLiveStreamDelegate: AnyObject {
+  func objectDetectorService(_ objectDetectorService: ObjectDetectorService,
+                             didFinishDetection result: ResultBundle?,
+                             error: Error?)
+}
+
+/**
+ This protocol must be adopted by any class that wants to take appropriate actions during  different stages of object detection on videos.
+ */
+protocol ObjectDetectorServiceVideoDelegate: AnyObject {
+ func objectDetectorService(_ objectDetectorService: ObjectDetectorService,
+                                  didFinishDetectionOnVideoFrame index: Int)
+ func objectDetectorService(_ objectDetectorService: ObjectDetectorService,
+                             willBeginDetection totalframeCount: Int)
+}
+
+
+// Initializes and calls the MediaPipe APIs for detection.
+class ObjectDetectorService: NSObject {
```

Comment:
Can it be `ObjectDetectorManager` to go along with `CameraFeedManager`? I notice that you put it into a folder called `Managers`.

Reply:
I didn't want to create another folder. Hence put the service in there. Will it be more appropriate to rename CameraFeedManager to CameraFeedService? Since the app is not all about live camera anymore, service sounds better since it handles only functionality specific to one part of the app. Manager sounds more global.

Reply:
`CameraFeedService` sgtm